src/prog_models/sim_result.py

```diff
@@ -72,7 +98,21 @@ def index(self, other: dict, *args, **kwargs) -> int:
         Returns:
             int: Index of first sample where other occurs
         """
+        warn('index will be deprecated after version 1.5 of ProgPy. The function will be renamed, indexofdata, and users may begin using it under this name now.',
+            DeprecationWarning, stacklevel=2)
+
         return self.data.index(other, *args, **kwargs)
+    def indexofdata(self, other: dict, *args, **kwargs) -> int:
```

Comment:
Should be index_of_data. See python naming standards: https://peps.python.org/pep-0008/#function-and-variable-names